py3status/modules/gazelle.py

```diff
@@ -0,0 +1,171 @@
+    """
+
+    # available configuration parameters
+    baseurl = 'https://mytracker.stuff'
+    cache_timeout = 60
+    cookiepath = '/var/tmp/gazelle_ratio_cookie.txt'
+    format = 'Ratio: {ratio}'
+    limit_required = None
+    limit_warning = 1
+    password = ''
+    username = ''
+
+    def _readable_size(self, size):
+        num = size
+        for unit in ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB']:
+            if abs(num) < 1024.0:
+                return "%3.1f %s" % (num, unit)
+            num /= 1024.0
+        return "%.1f%s" % (num, 'YiB')
+
+    def _login(self):
+        self.py3.request(
+                self.baseurl+"/login.php",
+                data={'username': self.username, 'password': self.password},
+                headers={'User-Agent': 'Mozilla/5.0'},
+                cookiejar=self._cookiejar
+                )
+        try:
+            self._index()  # just to try if login info are good
+        except RequestException:
+            self.py3.log(
+                    "Cannot login to %s" % self.baseurl,
+                    level=self.py3.LOG_ERROR
+                    )
+            raise
+        try:
+            self._cookiejar.save(ignore_discard=True)
+        except IOError:
+            self.py3.log(
+                    "Cannot write cookie at %s" % self.cookiepath,
+                    level=self.py3.LOG_WARNING
+                    )
+        except AttributeError:
+            pass  # That means that no cookie pass has been set
+        else:
+            self.py3.log(
+                    "Created cookie at %s" % self.cookiepath,
+                    level=self.py3.LOG_INFO
+                    )
+
+    def _index(self):
+        resp = self.py3.request(
+                self.baseurl+"/ajax.php",
+                params={'action': "index"},
+                headers={
+                    'User-Agent': 'Mozilla/5.0'
+                    },
+                cookiejar=self._cookiejar
+                )
+        return resp.json()
+
+    def post_config_hook(self):
+        if self.cookiepath:
+            self._cookiejar = MozillaCookieJar(self.cookiepath)
+            try:
+                self._cookiejar.load(ignore_discard=True)
+                self._index()  # just to try if saved cookie is good
+            except IOError:
+                self.py3.log(
+                        "Cannot read cookie at %s" % self.cookiepath,
+                        level=self.py3.LOG_INFO
+                        )
+                self._login()
+            except RequestException:
+                self.py3.log(
+                        "Cookie invalid or outdated. Re-logging.",
+                        level=self.py3.LOG_INFO
+                        )
+                self._login()
+        else:
+            self._cookiejar = CookieJar()
+
+    def gazelle(self):
+        index = self._index()
+        if index['status'] != "success":
+            msg = "Tracker failed to give me user info"
```

Comment:
(Guess) I think `index['status']` may fail here on no Internet / error.

Reply:
Well, guess I have to protect `self._index` then.